### PR TITLE
RestApiImp

### PR DESCRIPTION
**Pull request by hanusahani4**

### CODE REVIEW

**Review by manisharawat2310 — CHANGES_REQUESTED** [review]

File: springboot-backend/src/main/java/net/practice/restApi/controller/EmployeeController.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+@RestController
+@RequestMapping("/api/employees")
+public class EmployeeController {
+	@Autowire
+	private EmployeeService employeeService;
+
+	@PostMapping()
+	public ResponseEntity<Employee> saveEmployee(@RequestBody Employee employee){
+		return new ResponseEntity<Employee>(employeeService.saveEmployee(employee), HttpStatus.CREATED);
```
Comment (wowabhishek) [inline springboot-backend/src/main/java/net/practice/restApi/controller/EmployeeController.java:27]: Use static status code. 
like new ResponseEntity<Employee>(employeeService.saveEmployee(employee), CREATED);

File: springboot-backend/src/main/java/net/practice/restApi/controller/EmployeeController.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+	}
+	
+	@GetMapping
+	public List<Employee> getAllEmployees(){
+		return employeeService.getAllEmployees();
```
Comment (wowabhishek) [inline springboot-backend/src/main/java/net/practice/restApi/controller/EmployeeController.java:32]: if ResponseEntity for all method in controller

File: springboot-backend/src/main/java/net/practice/restApi/controller/EmployeeController.java [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+												  ,@RequestBody Employee employee){
+		return new ResponseEntity<Employee>(employeeService.updateEmployee(employee, id), HttpStatus.OK);
+	}
+	
+
+	@DeleteMapping("{id}")
+	public ResponseEntity<String> deleteEmployee(@PathVariable("id") long id){
+		
+
+		employeeService.deleteEmployee(id);
```
Comment (wowabhishek) [inline springboot-backend/src/main/java/net/practice/restApi/controller/EmployeeController.java:53]: if Id is not found then your API will always return successfully deleted employee based on id.

File: springboot-backend/src/main/java/net/practice/restApi/model/Employee.java [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.OneToOne;
+import javax.persistence.Table;
+
+import lombok.Data;
+
+@Entity
+@Table(name = "employees")
```
Comment (wowabhishek) [inline springboot-backend/src/main/java/net/practice/restApi/model/Employee.java:15]: table name should be singular

File: springboot-backend/src/main/java/net/practice/restApi/service/impl/EmployeeServiceImpl.java [hunk 5]
```diff
@@ -0,0 +1,67 @@
...
+		
+	}
+
+	@Override
+	public Employee updateEmployee(Employee employee, long id) {
+		
+		Employee existingEmployee = employeeRepository.findById(id).orElseThrow(
+				() -> new ResourceNotFoundException("Employee", "Id", id)); 
+		
+		existingEmployee.setFirstName(employee.getFirstName());
```
Comment (wowabhishek) [inline springboot-backend/src/main/java/net/practice/restApi/service/impl/EmployeeServiceImpl.java:52]: for setter should be use mapper package

File: springboot-backend/.mvn/wrapper/MavenWrapperDownloader.java [hunk 6]
```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright 2007-present the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
```
Comment (wowabhishek) [inline springboot-backend/.mvn/wrapper/MavenWrapperDownloader.java:6]: what is the use of this class

File: springboot-backend/mvnw.cmd [hunk 7]
```diff
@@ -0,0 +1,182 @@
+@REM ----------------------------------------------------------------------------
+@REM Licensed to the Apache Software Foundation (ASF) under one
```
Comment (wowabhishek) [inline springboot-backend/mvnw.cmd:2]: what is use of this class

**Review by wowabhishek — CHANGES_REQUESTED** [review]

**Pull request closed without merging**